src/lib/formats/os9_dsk.cpp

```diff
@@ -76,7 +76,7 @@ int os9_format::identify(util::random_read &io, uint32_t form_factor, const std:
 	int const type = find_size(io, form_factor, variants);
 
 	if (type != -1)
-		return FIFID_SIZE;
+		return FIFID_STRUCT | FIFID_SIZE;
```

Comment:
Please don't lie in those returns.  You're not checking anything of the structure of the file, only its size.

If you want to give a format priority over the others in sych a context, use FIFID_HINT.